src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -325,40 +333,41 @@ private static IEnumerable<PackageVulnerabilityMetadata> GetPackageVulnerabiliti
             return Enumerable.Empty<PackageVulnerabilityMetadata>();
         }
 
-        private static void WarnForHttpSources(
-            ListPackageArgs listPackageArgs,
-            ListPackageProjectModel projectModel)
+        private static List<PackageSource> GetInsecureHttpSources(IReadOnlyList<PackageSource> packageSources)
         {
-            var httpPackageSources = new List<PackageSource>();
-
-            AddHttpPackageSources(listPackageArgs.PackageSources, httpPackageSources);
-            AddHttpPackageSources(listPackageArgs.AuditSources, httpPackageSources);
+            List<PackageSource> httpPackageSources = null;
 
-            if (httpPackageSources.Count == 0)
+            foreach (PackageSource packageSource in packageSources)
             {
-                return;
+                if (packageSource.IsHttp && !packageSource.IsHttps && !packageSource.AllowInsecureConnections)
+                {
+                    httpPackageSources ??= new(capacity: packageSources.Count);
+                    httpPackageSources.Add(packageSource);
+                }
             }
 
-            string warningMessage = httpPackageSources.Count == 1
-                ? string.Format(CultureInfo.CurrentCulture, Strings.Warning_HttpServerUsage, "list package", httpPackageSources[0])
-                : string.Format(CultureInfo.CurrentCulture, Strings.Warning_HttpServerUsage_MultipleSources, "list package", Environment.NewLine + string.Join(Environment.NewLine, httpPackageSources.Select(e => e.Name)));
-
-            projectModel.AddProjectInformation(ProblemType.Warning, warningMessage);
+            return httpPackageSources ?? new();
         }
 
-        private static void AddHttpPackageSources(IEnumerable<PackageSource> packageSources, List<PackageSource> httpPackageSources)
+        private static void GetHttpSourceError(List<PackageSource> httpSources, out string error)
```

Comment:
Do we need the out param here?
Maybe just use string return type for the method GetHttpSourceError. 

You can choose how to handle 0 sources, like for example throw or just return null if there are 0 http surces.